### ISSUE TITLE
Support backward/gradient decomposition tracing

### ISSUE THREAD

**Issue by stmcgovern** [stmcgovern/decomposition-magician#2]
## Problem

The tool only traces forward decompositions. For ops with custom backward formulas (defined in `derivatives.yaml`), there's no way to see what the backward pass decomposes into.

For example, `logsumexp` has:
- Forward: `amax` → `sub` → `exp` → `sum` → `log` → `add`
- Backward (`logsumexp_backward`): `unsqueeze_multiple` → `sub` → `exp` → `conj` → `mul`

The backward graph is invisible to the tool.

## Proposed solution

A `--backward` flag that traces the backward decomposition:

```
$ decomp-magician logsumexp --backward

aten.logsumexp.default  backward
├── aten.unsqueeze.default  x2
├── aten.sub.Tensor
├── aten.exp.default
├── aten.conj.default
└── aten.mul.Tensor
```

Implementation approach: create a simple tensor with `requires_grad=True`, call the op, call `.backward()`, and record ops dispatched during the backward pass (similar to how the forward trace works).

## Use case

When adding autograd support for a new op variant (like `logsumexp.default` in pytorch/pytorch#160585), you need to verify the backward formula decomposes correctly. Currently this requires manual Python scripting.

**Comment by stmcgovern** [comment]
Implemented in v0.5.0 — `--backward` flag traces gradient ops.
